src/gt4py/cartesian/gtc/dace/oir_to_dace.py

```diff
@@ -162,14 +164,15 @@ def visit_Stencil(self, node: oir.Stencil, **kwargs):
             dim_strs = [d for i, d in enumerate("IJK") if decl.dimensions[i]] + [
                 f"d{d}" for d in range(len(decl.data_dims))
             ]
+            shape = ctx.make_shape(decl.name)
+            strides = [dace.symbolic.pystr_to_symbolic(f"__{decl.name}_{dim}_stride") for dim in dim_strs]
+            total_size = dace.data._prod([dace.symbolic.pystr_to_symbolic(s) for s in shape])
             ctx.sdfg.add_array(
                 decl.name,
-                shape=ctx.make_shape(decl.name),
-                strides=[
-                    dace.symbolic.pystr_to_symbolic(f"__{decl.name}_{dim}_stride")
-                    for dim in dim_strs
-                ],
+                shape=shape,
+                strides=strides,
```

Comment:
We should move stride/shape/total_size comptuation in a single function as to remove the code duplication. Probably in `cartesian/backend/dace_backend.py` or on a dedicated memory helper file in `cartesian/gtc/dace/`

Reply:
sure, totally agree

Reply:
add function in cartesian/gtc/dace/utils.py to calculate the array strides and array total size in order to reduce the duplication of code